core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1003,6 +1004,16 @@ object GroupMetadataManager {
 
   private val MEMBER_METADATA_V2 = MEMBER_METADATA_V1
 
+  private val MEMBER_METADATA_V3 = new Schema(
```

Comment:
We did get some flack in 2.1 for a change to this format. The problem is that we cannot downgrade once the new format is in use. We will probably have to mention this in the upgrade notes at a minimum. Unfortunately I don't see any great options at the moment to avoid this. Perhaps we should just switch to json.

Reply:
I see, we could discuss offline some time for a holistic solution.